R/utils.R

```diff
@@ -259,85 +259,560 @@ rowcol_to_dim <- function(row, col) {
   # we will always return something like "A1"
   stringi::stri_join(min_col, min_row)
 }
-#' @rdname dims_helper
-#' @param ... construct dims arguments, from rows/cols vectors or objects that can be coerced to data frame
+
+# It is inspired heavily by `rlang::arg_match(multi = TRUE)` and `base::match.arg()`
```

Comment:
Only inspired because it's just that it doesn't allow partial matching compared to base R.
rlang is licensed with MIT too. I can remove heavily though

Reply:
also, I am using this to validate arguments and produce an informative error message for users. And to do this, I use 30 lines, I felt like it made the already heavy `wb_dims()` function heavier. Also, it can be used in other places, where a user should supply values or arguments that need to be in a certain set

Reply:
I just wanted to make sure that we are not directly copying a function. Otherwise we should have added some license header for `rlang`. If it's just inspired that's fine by me.

People use to say "the pathway to hell is plastered with good intentions". Now we have this function here, that isn't used anywhere else but in `wb_dims()` and `wb_dims()` is already bloated. And we need changes to col2int() and assert_class() only to create a function constructing a simple A1 cell reference. And please be aware, I'm not saying these changes aren't justified, but I want to make sure that they are not just introduced because of either laziness (now or by previous developers) or lack of knowledge of the `openxlsx2` internals. Because that's what happened to `openxlsx`, where e.g. multiple functions for `col2int()` were introduced over the years (even in R and C++), because everyone always assumed that they were needed and nobody considered that maybe the function they were writing was already around.

Reply:
Hi. 

The changes to col2int are mostly related to base R's inconsistencies...

as.integer(NULL) returns integer(0), but nrow(NULL) returns NULL.

NULL is much easier to handle than integer(0) hence the small change.

regarding the now 3 functions only used by wb_dims(), I wouldn't mind putting them back inside wb_dims() if you'd like. 
sometimes, it's just that more scoped functions are easier.

I really don't like base R error messages, and the fact that it's failing silently sometimes.
```r
match.arg(c("rows", "whatever", "cols", "col"), c("cols, "rows"), several.ok = T)
"rows" "cols" "cols" 
```
hence the need for the wrapper..

assert_class changes are also because when I struggle at first, an internal error was thrown by assert_class(), because `sx` was `character(0)`, which was reallly hard to debug...